task/dispatcher_test.go

```diff
@@ -24,14 +25,17 @@ func TestNewDispatcher(t *testing.T) {
 
 	t.Run("NewFailed", func(t *testing.T) {
 		patches = testp2p.P2pNewPubSubs(patches, nil, errors.New(t.Name()))
-		_, err := NewDispatcher(nil, nil, "", "", "", 0)
+		_, err := NewDispatcher(clock.New().Ticker(3*time.Second), nil, nil, "", "", "", 0)
```

Comment:
the system "time.NewTicker" is a func, maybe we can use "github.com/bytedance/mockey" to mock it return value. 
for we have other resource func call, like "ethclient.Dial", "http.Get". we can use mockey to mock all of them. and avoid introducing too many dependencies